### PR TITLE
Ensure CenterCrop indices are integers

### PR DESCRIPTION
**Pull request by dribnet**
On my system when adding an augmentation CenterCrop, I was getting
tons of warnings:

  points_src: torch.Tensor = torch.tensor(
.../kornia/augmentation/random_generator/random_generator.py:713:
DeprecationWarning:
 an integer is required (got type float).  Implicit conversion to
integers using __int__ is deprecated, and may be removed in a future
version of
Python.

this commit forces the start_x start_y indices to be ints
to remove this warning.

### Description
A few sentences describing the changes proposed in this pull request.

### Status
**Ready/Work in progress/Hold**

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or new feature that would cause existing functionality to change)
- [ ] New tests added to cover the changes
- [ ] Docstrings/Documentation updated

### PR Implementer
This is a small checklist for the implementation details of this PR.

If there are any questions regarding code style or other conventions check out our
[summary](https://github.com/kornia/kornia/blob/master/CONTRIBUTING.rst).

- [ ] Did you discuss the functionality or any breaking changes before ?
- [ ] **Pass all tests**: did you test in local ? `make test`
- [ ] Unittests: did you add tests for your new functionality ?
- [ ] Documentations: did you build documentation ? `make build-docs`
- [ ] Implementation: is your code well commented and follow conventions ? `make lint`
- [ ] Docstrings & Typing: has your code documentation and typing ? `make mypy`
- [ ] Update notebooks & documentation if necessary

### KorniaTeam
<details>
  <summary>KorniaTeam workflow</summary>

  - [ ] Assign correct label
  - [ ] Assign PR to a reviewer
  - [ ] Does this PR close an Issue? (add `closes #IssueNumber` at the bottom if
        not already in description)

</details>

### Reviewer
<details>
  <summary>Reviewer workflow</summary>

  - [ ] Do all tests pass? (Unittests, Typing, Linting, Documentation, Environment)
  - [ ] Does the implementation follow `kornia` design conventions?
  - [ ] Is the documentation complete enough ?
  - [ ] Are the tests covering simple and corner cases ?

</details>

### CODE REVIEW

File: kornia/augmentation/random_generator/random_generator.py [hunk 1]
```diff
@@ -693,8 +693,8 @@ def center_crop_generator(
     src_h_half = src_h / 2
     src_w_half = src_w / 2
 
-    start_x = src_w_half - dst_w_half
-    start_y = src_h_half - dst_h_half
+    start_x = int(src_w_half - dst_w_half)
+    start_y = int(src_h_half - dst_h_half)
```
Comment (edgarriba) [inline kornia/augmentation/random_generator/random_generator.py:697]: @shijianjian will this affect to the subpixel precision when we crop with sampling ?

Comment (shijianjian) [reply kornia/augmentation/random_generator/random_generator.py:697]: Probably fine. But @dribnet, can I know the kornia and torch version you are using?

https://github.com/kornia/kornia/blob/d18c392a666b4d242413705848e7500cc7280046/kornia/augmentation/random_generator/random_generator.py#L713

The error line you referred is actually an empty line there.

**Comment by dribnet** [comment]
```bash
# Name                  Version                         Build    Channel
kornia                    0.5.6                  pyhd8ed1ab_0    conda-forge
pytorch                   1.9.0   py3.8_cuda10.2_cudnn7.6.5_0    pytorch
```

**Review by edgarriba — APPROVED** [review]

**Review by shijianjian — APPROVED** [review]

**Comment by edgarriba** [comment]
Thanks for the contribution @dribnet
